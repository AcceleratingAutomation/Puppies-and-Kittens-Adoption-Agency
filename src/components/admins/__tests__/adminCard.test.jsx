import { render, screen } from "@testing-library/react";
import { adminProps, getAdminCard } from "./adminsData";

describe("Admin Card", () => {
  beforeEach(() => {
    render(getAdminCard());
  });

  test("matches snapshot comparison", () => {
    const { asFragment } = render(getAdminCard());
    expect(asFragment()).toMatchSnapshot();
  });

  test("displays admin name", () => {
    expect(screen.getByText(adminProps.name)).toBeInTheDocument();
  });

  test("displays email", () => {
    expect(screen.getByText(adminProps.email)).toBeInTheDocument();
  });

  test("displays number of current rescues", () => {
    expect(
      screen.getByText(`Current Rescues: ${adminProps.numCurrentRescues}`),
    ).toBeInTheDocument();
  });

  test("displays number of total rescues", () => {
    expect(
      screen.getByText(`Total Rescues: ${adminProps.numTotalRescues}`),
    ).toBeInTheDocument();
  });

  test("displays number of household pets", () => {
    expect(
      screen.getByText(`Household Pets: ${adminProps.numHouseholdPets}`),
    ).toBeInTheDocument();
  });

  test("displays number of favorites", () => {
    expect(
      screen.getByText(`Favorites: ${adminProps.favorite.length}`),
    ).toBeInTheDocument();
  });

  test("displays view details button", () => {
    expect(screen.getByText("VIEW DETAILS")).toBeInTheDocument();
  });
});
