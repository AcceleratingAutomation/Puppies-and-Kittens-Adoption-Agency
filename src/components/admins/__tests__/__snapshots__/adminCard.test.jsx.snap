// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Card matches snapshot comparison 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root rescue makeStyles-summaryCard-1 MuiPaper-elevation2 MuiPaper-rounded"
  >
    <img
      alt="John Doe's picture"
      class="makeStyles-responsiveImage-3"
      src="/images/users/placeholder-users.webp"
      style="border-radius: 25%; width: 15rem; height: 18rem; object-fit: cover;"
    />
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4"
        >
          John Doe
        </h4>
      </div>
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        john.doe@example.com
      </h5>
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        Current Rescues: 5
      </h5>
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        Total Rescues: 10
      </h5>
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        Household Pets: 3
      </h5>
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        Favorites: 3
      </h5>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-xs-12"
      />
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-xs-12"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-muiButton-2 MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            VIEW DETAILS
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
